[CAP-64-140] Google calendar import functionality

Ava-Samimi opened the pull request:
### 📌 Feature: Google Calendar Integration

### 🔍 Summary
- Added functionality to import Google Calendar events into the app.
- Implemented Google Sign-In to request user permissions.
- Fetched and translated calendar events into a usable format.

### 🛠 Changes Made
- Created GoogleSignInActivity.java for authentication.
- Created GoogleCalendarService.java to fetch calendar events.
- Added an "Import Calendar" button in activity_calendar.xml.
- Updated MainActivity.java to start GoogleSignInActivity.

### ✅ Testing
- Verified Google Sign-In works correctly.
- Confirmed calendar events are fetched and parsed.
- Ensured a smooth navigation experience.

### 🔗 Next Steps
- Sync with UI team to display events properly.

Ava-Samimi committed:
feat: google calendar import functionality

Ava-Samimi committed:
fix: dependency conflict

Ava-Samimi commented:
There was an issue with the build. Added these two to the Gradle to resolve same dependencies in multiple places:

android {
    packagingOptions {
        resources {
            excludes += "META-INF/DEPENDENCIES"
            excludes += "META-INF/NOTICE"
            excludes += "META-INF/LICENSE"
        }
    }
}


AND

dependencies {
    implementation("com.google.api-client:google-api-client-android:1.33.0") {
        exclude(group = "org.apache.httpcomponents")
    }
    implementation("com.google.http-client:google-http-client-gson:1.42.3") {
        exclude(module = "httpclient")
    }
}





problem solved1

Kayram2710 commented on app/src/main/res/layout/activity_calendar_import.xml:
Ok so just to clarify this xml file is not necessary as the UI for the calendar, its already being implemented in another PR

Kayram2710 commented on app/src/main/java/minicap/concordia/campusnav/screens/MainActivity.java:
```diff
@@ -28,6 +28,10 @@ protected void onCreate(Bundle savedInstanceState) {
+        // to handle Google Calendar integration
+        Intent calendarImportIntent = new Intent(this, CalendarImportActivity.class);
+        startActivity(calendarImportIntent);
+
```
Its really bad practice leaving this in there, its rerouting the entire workflow of the app into the activity and not letting the app launch.

Kayram2710 replied:
And for some reason its causing a crash once its launched. This might be because the Activity is erroneous.

Kayram2710 commented on app/src/main/java/minicap/concordia/campusnav/screens/CalendarImportActivity.java:
This class could use some tests. And also it crashed when launched. The logic is a bit too taylored for the xml you created.

Kayram2710 reviewed (changes requested):
As it is the branch needs more work to further implement the requirements, solid work creating the baseline for the API request, but theres a lot of development malpractices to look out for. To complete this PR please work with @damle22 since he also worked on the front end for it.

Ava-Samimi committed:
feat: make google calendar import process start through a button click from the main activity

Ava-Samimi committed:
fix: removed the import button from main activity

damle22 reviewed (commented):
We went over the PR together with @Ava-Samimi on a call. She will address the issues today by pulling from the CalendarUI branch that I made and using the components of the UI to complete her task.

Ava-Samimi committed:
Merge branch 'CAP-57-141-ImplementUICalendar' into CAP-140-GoogleCalendar

Ava-Samimi committed:
feat: remove calendar data fetch

link UI button to activity

Ava-Samimi committed:
fix

Ava-Samimi commented:
📌 Feature: Google Calendar Integration

🛠 Changes Made:

- merged the CAP-57-141 branch into the CAP-140 branch 
- removed the calendar data fetch logic
- linked the UI button to the calendar import logic
- removed import button from main activity
- modified logic so it does not crash
- removed unnecessary xml file 

🔗 Next Steps
Sync with UI team to display events properly. [damle22](https://github.com/damle22)

Ava-Samimi committed:
Merge branch 'main' into CAP-140-GoogleCalendar

Ava-Samimi committed:
fix

Ava-Samimi committed:
fix

Ava-Samimi committed:
fix

damle22 reviewed (commented):
![image](https://github.com/user-attachments/assets/c1dbd957-a50c-4c3e-a3b7-be42eebf6b57)

When I run it and I try connecting to the Goggle Services by using a Gmail Account it fails straight away. It says 'Google Sign In Failed'. Did you test the functionality on your end before committing? Because it does not meet the requirement as of now.

Ava-Samimi committed:
feat: add google oauth client ID for the goole signin success

Ava-Samimi committed:
feat: setup google auth clien

damle22 committed:
debug WEB_CLIENT_ID

Now we can access the Google and GoogleCalendar of the user email address. It is now functional.

damle22 committed:
fixed security issue 1

damle22 committed:
fix to make sonarQube security test pass

damle22 committed:
Delete activity_calendar_import.xml

damle22 commented:
I just committed to this PR to make it pass the requirements. We can now access Google and Google Calendar permission. Should be good to good.

Kayram2710 committed:
Merge branch 'main' into CAP-140-GoogleCalendar

Kayram2710 committed:
Fixed merge

Kayram2710 reviewed (approved):
After going over everything it seems the functionality works, however I am having a bug on my end with the sign in failing which i was able to test was only happening for me
